File: FE/src/domains/Post.ts
import User from '../domains/User'
import PostType from '../enums/PostType'

class Post {
    id?: string;
    title?: string;
    picture?: Uint8Array;
    content?: string;
    description?: string;
    creator?: User;
    postType?: PostType;
    tags?: string[]
  
    constructor(
      id?: string,
      title?: string,
      picture?: Uint8Array,
      content?: string,
      description?: string,
      creator?: User,
      postType?: PostType,
      tags?: string[]
    ) {
      this.id = id;
      this.title = title;
      this.picture = picture;
      this.content = content;
      this.description = description;
      this.creator = creator;
      this.postType = postType;
      this.tags = tags;
    }
  }

  export default Post

File: BE/src/main/java/be/ytbe/domain/Post.java
package be.ytbe.domain;

import be.ytbe.domain.enumeration.PostType;
import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class Post {
    private String id;
    private String title;
    private byte[] picture;
    private Object content;
    private String description;
    private User creator;
    private PostType postType;
}
